allows using custom CA bundle for build process

xificurC opened the pull request:
**What's this PR for?
Allowing to build the docker Zulip image with a custom CA bundle

**Testing Plan:** <!-- How have you tested? -->
Run `docker-compose build` in zulip/docker-zulip

timabbott commented on scripts/lib/setup_venv.py:
```diff
@@ -252,6 +252,37 @@ def setup_virtualenv(target_venv_path, requirements_file, virtualenv_args=None,
+    else:
+        with open(conffile, 'x') as new:
+            new.write('[global]\n')
+            new.write("cert = {}\n".format(os.environ.get('CUSTOM_CA_CERTIFICATES')))
```
This is a lot of code to get right.  `pip.conf` is in the standard `.ini` format supported by Python's ConfigParser library, so we could use that, but I think this function can be replaced with a single call to `crudini`, which we already install as one of our `apt` dependencies and thus should be available here.

xificurC replied:
I searched for a library but didn't land on ConfigParser or anything alike, thanks for the tips. I will check `crudini` and rewrite as needed.

> This is a lot of code to get right

Indeed, I hand-tested each branch

timabbott commented:
@xificurC I posted a quick comment on how to remove most of the code in this implementation.  

Otherwise, this looks reasonable, though we generally don't accept environment variables as input to the install process in production wherever possible.  So we might want to think about making this a command-line argument to `install`.

xificurC commented:
Which install script do you mean, [this one](https://github.com/zulip/zulip/tree/master/scripts/lib/install)? I'll need to check where to propagate the CLI argument but shouldn't be much trouble to rewrite I guess

xificurC commented:
Hi @timabbott , I changed the pipconf setup to use `crudini` and added a CLI flag for a custom certiciate bundle. Not sure if you wanted to propagate that value downwards as an argument as well or if using the current method (exporting the environment variable inside the install script) is what you had in mind?

timabbott commented:
I reviewed this and made a few style tweaks, and then repushed to your PR branch (I'd like to see Travis CI, our production installation test suite, pass before merging)

timabbott commented:
Merged, thanks @xificurC!

xificurC commented:
@timabbott thanks for taking the time to go through all the PRs here and in docker-zulip
